\documentclass[letterpaper]{article}

\usepackage{natbib,alifeconf}  %% The order is important
\usepackage{amssymb, amsmath, amsthm}
\newtheorem{proposition}{Proposition}


% *****************
%  Requirements:
% *****************
%
% - All pages sized consistently at 8.5 x 11 inches (US letter size).
% - PDF length <= 8 pages for full papers, <=2 pages for extended
%    abstracts (not including citations).
% - Abstract length <= 250 words.
% - No visible crop marks.
% - Images at no greater than 300 dpi, scaled at 100%.
% - Embedded open type fonts only.
% - All layers flattened.
% - No attachments.
% - All desired links active in the files.

% Note that the PDF file must not exceed 5 MB if it is to be indexed
% by Google Scholar. Additional information about Google Scholar
% can be found here:
% http://www.google.com/intl/en/scholar/inclusion.html.


% If your system does not generate letter format documents by default,
% you can use the following workflow:
% latex example
% bibtex example
% latex example ; latex example
% dvips -o example.ps -t letterSize example.dvi
% ps2pdf example.ps example.pdf


% For pdflatex users:
% The alifeconf style file loads the "graphicx" package, and
% this may lead some users of pdflatex to experience problems.
% These can be fixed by editing the alifeconf.sty file to specify:
% \usepackage[pdftex]{graphicx}
%   instead of
% \usepackage{graphicx}.
% The PDF output generated by pdflatex should match the required
% specifications and obviously the dvips and ps2pdf steps become
% unnecessary.


% Note:  Some laser printers have a serious problem printing TeX
% output. The use of ps type I fonts should avoid this problem.


\title{Dynamics of Random Fuzzy Networks}
\author{Carlos Gershenson$^{1,2}$, Hyobin Kim$^1$, Octavio Zapata$^1$
\mbox{}\\
$^1$Centro de Ciencias de la Complejidad , Universidad Nacional Aut\'onoma de M\'exico, M\'exico \\
$^2$Instituto de Investigaciones en Matem\'aticas Aplicadas y en Sistemas, Universidad Nacional Aut\'onoma de M\'exico, M\'exico \\
cgg@unam.mx\ \ \ 
hyobin.kim@c3.unam.mx\ \ \ 
octavio.zapata@c3.unam.mx} % email of corresponding author(s)

% For several authors from the same institution use the same number to
% refer to one address.
%
% If the names do not fit well on one line use
%         Author 1, Author 2 ... \\ {\Large\bf Author n} ...\\ ...
%
% If the title and author information do not fit in the area
% allocated, place \setlength\titlebox{<new height>} after the
% \documentclass line where <new height> is 2.25in



\begin{document}
\maketitle
%\tableofcontents
%\begin{abstract}
% Abstract length should not exceed 250 words
%\end{abstract}
Genes are the fundamental unit of inheritable information. 
A gene is a part of the genomic sequence that encodes how to produce (synthesise) either a protein or some RNA (a gene product). 
Gene product synthesis is called gene expression. 
Not all genes are expressed at the same time.
The expression of each gene is affected by the expression of other genes in a process called gene regulation. 
This gives rise to a network-like structure called genetic regulatory network. %, where there is one node $i$ for each gene, and one arc $i\to j$ if gene $i$ regulates the expression of gene $j$. 

Random Boolean networks were proposed by Kauffman in [] as models of genetic regulatory networks.
 The expression of each gene is represented using one bit: 1 represents the gene is expressed, and 0  the gene is not expressed.
Let $\{0,1\}^n$ be the set of all binary words of length $n$. A point $x=(x_1,\dots, x_n)\in \{0,1\}^n$ is called  a \emph{state} and $\{0,1\}^n$ is called the \emph{state-space}. Each index $i\in I:=\{1,\dots,n\}$ is identified with a gene.

A \emph{Boolean network} with parameters $n$ and $k$, $1\leq k\leq n,$ 
consists of a family 
$y=\{y(i):i\in I\}$
of
subsets $y(i)=\{i_1,\dots,i_k\}\subseteq I$, and a family $f=\{f_i: i\in I\}$ of functions $f_i\colon \{0,1\}^{k}\rightarrow \{0,1\}$. 
The $k$ genes in $y(i)$ are called the \emph{regulators} of $i$.
Notice that the number $k$ of regulators is the same for all genes.
A \emph{random Boolean network} is a Boolean network $(y,f)$, where   $y(i)$ and $f_i$ are chosen randomly and  independently of each other. %uniformly with probability $\binom{n}{k}^{-1}$, and each $f_i$  is chosen uniformly with probability $2^{-2^k}$, independently of each other. 

A \emph{directed graph} (or \emph{digraph}) $G$ consists of a finite set of elements $V(G)$ called vertices (or nodes), and a set of pairs of elements $E(G)\subseteq V(G)\times V(G)$ called directed edges (or arcs).  
For any arc $(u,v)\in E(G)$, we say that $v$ is a \emph{successor} of $u$. 
%The \emph{out-neighbourhood} of a node $i\in V(G)$ is the set $N^+(i)=\{j\in V(G):(j,i)\in E(G)\}$. 
The \emph{out-degree} of a node $v$ is the number %$d^+(x) =|\{y\in V(G):(x,y)\in E(G)\}|$ 
 of successors of $v$. 
%Similarly, the \emph{in-neighbourhood} of $i$ is $N^-(i)=\{j\in V(G):(i,j)\in E(G)\}$ and its \emph{in-degree} is $d^-(i) = |N^-(i)|$.
A \emph{directed pseudoforests} is a disjoint union of digraphs where each node has out-degree one. 
Every function $F\colon X\to X$,  from a set $X$ to itself, defines a directed pseudoforest with elements of $X$
as vertices, and for all $x,x'\in X$, a directed edge from $x$ to $x'$ if $F(x)=x'$.


Similarly, every Boolean network $(y,f)$ defines a directed pseudoforests
on the state-space $\{0,1\}^n$.
For each state $x=(x_1,\dots, x_n)\in \{0,1\}^n$ and each gene $i\in I$, let $x_{y(i)}:=(x_{i_1},\dots,x_{i_k})$. 
Given  a fuzzy network $(y,f)$,  we define  $F\colon \{0,1\}^n\ \to\ \{0,1\}^n$ by  
\[
F(x):=\big(
 f_1( x_{y(1)}), \dots,
 f_n( x_{y(n)})
\big),\quad x\in \{0,1\}^n.\qedhere
\]
Thus, we identify a Boolean network $(y,f)$ with the mapping $F$, and so with the directed pseudoforest on $\{0,1\}^n$.
  
  
Fuzzy networks are directed pseudoforests on the state-space of a gene regulatory network, where the expression of each gene is represented by one of $q\geq 2$ different values. 
That is fuzzy networks are directed pseudoforests on $\{0,1,\dots,q-1\}^n$.
A fuzzy network where $q=2$ is precisely a Boolean network.
 Fuzzy networks have been recently used to study some aspects related to the differentiation of cells from the immune system [], and also to explain the importance of certain gene products associated with the development of metabolic syndrome and type 2 diabetes [].  


 A \emph{fuzzy network} is a Boolean network $(y,f)$ with $f_i\colon \{0,1,\dots,q-1\}^k\rightarrow \{0,1,\dots,q-1\}$, $q\geq 2$, $i\in I$, such that
% \begin{align*}
%&f_i(x_{i_1},\dots,x_{i_k})=\\
% & \big(x_{i_1}\wedge f_i(x_{i_1},\dots,x_{i_k})\big) \vee \big(\lnot x_{i_1} \wedge f_i(\lnot x_{i_1},x_{i_2},\dots,x_{i_k})\big),
%  \end{align*}
 \begin{align*}
f_i(x_{i_1},\dots,x_{i_k})=& \big(x_{i_1}\wedge f_i(x_{i_1},\dots,x_{i_k})\big) \\
&\vee \big(\lnot x_{i_1} \wedge f_i(\lnot x_{i_1},x_{i_2},\dots,x_{i_k})\big),
  \end{align*}
  where $x\vee y:=\max\{x,y\}, x\wedge y:=\min\{x,y\}$, and $\lnot x:=q-1-x$.
  The number $q$ is called the \emph{base} of the network. 
Notice that in this model, the base  is the same for all genes.
Every fuzzy network with base $q$ is a directed pseudoforest on $\{0,1,\dots,q-1\}^n$. 

A \emph{random fuzzy network} is a fuzzy network $(y,f)$, where   $y(i)\subseteq I$ and $f_i\colon\{0,1,\dots,q-1\}^n\rightarrow\{0,1,\dots,q-1\}$ are chosen randomly and  independently of each other, for all $i\in I$.
Random fuzzy networks are a particular class of the random networks with multiple states of Sol\'e et. al. []. 


 A \emph{walk} in a digraph $G$ is a sequence of vertices  $v_1,v_2,\dots\in V(G)$, so that for all $j\geq 1, (v_j,v_{j+1})\in E(G)$.
A walk where all nodes are distinct is called a \emph{path}.
Paths are necessarily finite walks.   
A finite walk where the first and the last node are the same is called a \emph{cycle}. 
The number of distinct nodes in a cycle is the called the \emph{length} of the cycle.

%We say that a digraph $G$ is \emph{strongly connected} if there is a path in $G$ between any two distinct nodes in each direction, i.e. for all $u,v\in V(G)$ there is a path from $u$ to $v$ and a path from $u$ to $v$. 


The average number and length of state cycles for  random Boolean networks are two well-studied combinatorial parameters. 
 Let $C(n,q)$ be the average number of cycles  
 on a random fuzzy network with $n$ genes and base $q$, where the number $k$ of regulators per gene is fixed to some constant value. By a result of Kruskal [], we know that 
\[
C(n,q)\to\frac{1}{2}\log q^n + \bigg( \frac{\log 2 + C}{2} \bigg) + o(1),\quad n\to\infty,
\]
 where $C=0.5772\dots$ is Euler's constant.
  
 Samuelsson et al. [] gave a formula for $C(n, 2)$ which allows them to conclude that $C(n,2)$ grows faster than $n^{a}$, for any $a>0$.
  Our experimental research suggests that for  $b,q\geq 0$, if $b\leq q$ then
 \begin{equation}
 \label{eq:2}
 C(n,b)\leq C(n,q).
 \end{equation}
  In particular, $C(n,2)\leq C(n,q)$ for any $q\geq 2$.
So, equation \eqref{eq:2} would imply 
 \[
 \frac{n^{a}}{
    C(n,q)} \to 0,\qquad\qquad n\to\infty.
  \]
In words, the average number of cycles in a random fuzzy network  $C(n,q)$ grows faster than $n^{a}$, for any $a>0$. 
% Given $b,q\geq 0$, we have
% \[b^n\leq q^n,\qquad n\geq 1
% \] if $b\leq q$. 


% 
\newpage
 
\section{}




\footnotesize
%\bibliographystyle{apalike}
%\bibliography{example} % replace by the name of your .bib file


\end{document}
